Pipfile

```diff
@@ -10,12 +10,19 @@ pre-commit = "*"
 coverage = "*"
 pytest-cov = "*"
 codacy-coverage = "*"
-pipenv-setup = "*"
+iolite = {editable = true,path = "."}
+black = "==19.10b0"
+toml = "*"
 responses = "*"
+isort = "==4.3.21"
+pipenv-setup = "*"
 pytest-socket = "*"
 freezegun = "*"
 
 [packages]
 websockets = "*"
 environs = "*"
 requests = "*"
+
+[pipenv]
+allow_prereleases = true
```

Comment:
Black is only available as a pre-release package and pipenv does not install pre-releases without actively allowing it to do so. When installing it with `pipenv install black --dev --pre` the "allow_prereleases" part is automatically added by pipenv.
Once black becomes available as a major release, this can be removed again. I guess this can take some time, though: https://github.com/psf/black/issues/1746

Reply:
What's the need for `black`? I've set it up as a [pre-commit ](https://pre-commit.com/) hook which you can install via `pre-commit install` or just run manually with `pre-commit run --all`? 

Which is also called on CI

Reply:
Having black, isort, and flake8 installed in the environment, I use file watchers to run them on every file save. I prefer this setup to running the pre-commit hook once in a while, because I immediately see the changes made by black and isort. Also, this way I can fix the flake8 warnings right away.

I could probably also just run the pre-commit hook, this feels slightly slower, though (and runs on all files, not just the file I am working on).

So I added the scripts as dev dependencies, cause I though it does no harm  🙂

Reply:
Okays - that makes sense.

I think pre-commit runs on the diff by default but you can also tell it to run on all changes.

We can always remove the dev dependencies if it becomes too much.